### ISSUE TITLE
rollup superagent breaks tests

### ISSUE THREAD

**Issue by leeyeh** [rollup/rollup-plugin-commonjs#56]
Upgrading superagent to version 1.8 broke the tests of my app.
```
Uncaught TypeError: Cannot set property '_timeout' of undefined
```

related change:
```javascript
exports.clearTimeout = function _clearTimeout(){
  this._timeout = 0;
  clearTimeout(this._timer);
  return this;
};
```
https://github.com/visionmedia/superagent/blob/93b060790e78b6ca088aec57cb3168ee5e248007/lib/request-base.js#L12-L16

which was rolled up to:
```javascript
var requestBase = __commonjs(function (module, exports) {
//...
exports.clearTimeout = function _clearTimeout(){                                                                                                    
    this._timeout = 0;
    clearTimeout$1(this._timer); // expected: clearTimeout(this._timer);
    return this;
};
//...
});
var clearTimeout$1 = requestBase.clearTimeout;
```

**Comment by eventualbuddha** [comment]
Can you provide either a repository that exhibits this behavior or an example on http://rollupjs.org?

**Comment by leeyeh** [comment]
https://github.com/leeyeh/rollup-commonjs-plugin-issue-56

```
npm test
```

**Comment by eventualbuddha** [comment]
Okay, it took me a while to see the problem, but it seems that this plugin is clobbering globals. Your `request-base.js` file is being rewritten by this plugin to this:

```js
var requestBase = __commonjs(function (module, exports) {
// https://github.com/visionmedia/superagent/blob/93b060790e78b6ca088aec57cb3168ee5e248007/lib/request-base.js

exports.clearTimeout = function _clearTimeout(){
  this._timeout = 0;
  clearTimeout(this._timer);
  return this;
};
});

export default (requestBase && typeof requestBase === 'object' && 'default' in requestBase ? requestBase['default'] : requestBase);
export var clearTimeout = requestBase.clearTimeout;
```

That's not correct since it is creating a local binding for `clearTimeout` where there previously was none. That wouldn't be a problem if there wasn't already a reference to a global `clearTimeout`. Therefore, I think this should be the actual transformed code when an exported name conflicts with an implied global:

```js
var requestBase = __commonjs(function (module, exports) {
// https://github.com/visionmedia/superagent/blob/93b060790e78b6ca088aec57cb3168ee5e248007/lib/request-base.js

exports.clearTimeout = function _clearTimeout(){
  this._timeout = 0;
  clearTimeout(this._timer);
  return this;
};
});

export default (requestBase && typeof requestBase === 'object' && 'default' in requestBase ? requestBase['default'] : requestBase);
var clearTimeout$1 = requestBase.clearTimeout;
export { clearTimeout$1 as clearTimeout };
```

@TrySound?

**Comment by dtothefp** [comment]
@eventualbuddha having the same issue here....

@leeyeh  are you using the `node-resolve` plugin with `{browser: true}` options? I had a different compilation issue if I didn't use this option, but now we have the same `clearTimeout` global clobbering issue. Would be great to find a fix!!

@bethcodes

**Comment by leeyeh** [comment]
@dtothefp yes.
superagent@1.7 works fine for me.

**Comment by Rich-Harris** [comment]
Hi, sorry for the long, long wait to get this resolved. Fairly certain it was the same bug as #84, which was fixed by #107. If you update this plugin it should work fine – will close this but reopen if it isn't fixed
